test/OpenTelemetry.Tests/Shared/SkipUnlessEnvVarFoundFactAttribute.cs

```diff
@@ -0,0 +1,44 @@
+// <copyright file="SkipUnlessEnvVarFoundFactAttribute.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using Xunit;
+
+namespace OpenTelemetry.Tests
+{
+    internal class SkipUnlessEnvVarFoundFactAttribute : FactAttribute
+    {
+        public SkipUnlessEnvVarFoundFactAttribute(string environmentVariable)
+        {
+            if (string.IsNullOrEmpty(GetEnvironmentVariable(environmentVariable)))
+            {
+                this.Skip = $"Skipped because {environmentVariable} environment variable was not configured.";
+            }
+        }
+
+        public static string GetEnvironmentVariable(string environmentVariableName)
```

Comment:
I was going to say you could call `SkipUnlessEnvVarFoundTheoryAttribute.GetEnvironmentVariable` to achieve some code reuse but then you would have to also include that file in the project which kind of sucks. Probably the duplication is better. So I guess what I'm saying is, nice job 😄

Reply:
Yea I had some similar thoughts then opted for duplication. We could play around with organizing things like this into a "xUnit extensions" namespace and then you could do something like

```xml
<Compile Include="$(RepoRoot)\test\OpenTelemetry.Tests\xunit\**" />
```